From 0afb9dc7dffae4a332393c4bff450eb072b02929 Mon Sep 17 00:00:00 2001
From: Guoqing Jiang <guoqing.jiang@suse.com>
Date: Wed, 19 Jan 2022 17:26:27 +0800
Subject: [PATCH] fix compile issue of hinic_ethtool.c
Patch-mainline: Never, fixes out-of-tree driver
References: patch dependencies

Since {get,set}_ringparam has changed their parameter in the commit
("ethtool: extend ringparam setting/getting API with rx_buf_len"),
the hinic driver need to be modified as well.

Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/net/ethernet/huawei/hinic/hinic_ethtool.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/huawei/hinic/hinic_ethtool.c b/drivers/net/ethernet/huawei/hinic/hinic_ethtool.c
index 28c7a34d204a..26d695684cfb 100644
--- a/drivers/net/ethernet/huawei/hinic/hinic_ethtool.c
+++ b/drivers/net/ethernet/huawei/hinic/hinic_ethtool.c
@@ -905,7 +905,9 @@ static int hinic_nway_reset(struct net_device *netdev)
 }
 
 static void hinic_get_ringparam(struct net_device *netdev,
-				struct ethtool_ringparam *ring)
+				struct ethtool_ringparam *ring,
+				struct kernel_ethtool_ringparam *kernel_ring,
+				struct netlink_ext_ack *extack)
 {
 	struct hinic_nic_dev *nic_dev = netdev_priv(netdev);
 
@@ -931,7 +933,9 @@ static void hinic_update_qp_depth(struct hinic_nic_dev *nic_dev,
 }
 
 static int hinic_set_ringparam(struct net_device *netdev,
-			       struct ethtool_ringparam *ring)
+			       struct ethtool_ringparam *ring,
+			       struct kernel_ethtool_ringparam *kernel_ring,
+			       struct netlink_ext_ack *extack)
 {
 	struct hinic_nic_dev *nic_dev = netdev_priv(netdev);
 	u16 new_sq_depth, new_rq_depth;
-- 
2.26.2

